docker/Dockerfile-backend

```diff
@@ -0,0 +1,29 @@
+FROM alpine:3.8 as builder
+
+RUN apk update
+RUN apk add build-base \
+            cmake \
+            curl-dev \
+            git \
+            go \
+            linux-headers \
+            perl
+
+WORKDIR /root/dronecode_sdk
+
+RUN git clone https://github.com/dronecode/dronecodesdk --recursive .
```

Comment:
Can we use an `ARG` here to determine the branch/tag? we could also do a shallow clone with depth 1

Example:
```dockerfile
...
ARG branch=master
RUN git clone --branch $branch --depth 1  https://github.com/dronecode/dronecodesdk --recursive .
...
```

This will make it easier to do builds per releases on CI

```bash
docker build --build-arg branch="v0.14.1" -f Dockerfile-backend -t dronecode/dronecode-sdk-base .
```

Although I'm not sure if this messes up your multistage build and how caching works due to the ARG

Reply:
I am not sure about the ARG and the deployment on docker hub. I've never automated that... But if it helps I'm open to it.

About the shallow clone, that won't work because of the submodules.

Reply:
Actually I'm thinking that instead of an `ARG` and `git clone`, we should probably `COPY` the sources. Because that will get built by Travis, and it has a clone of the repo already :+1:.

Reply:
What about the newer version? Note that when building from the `docker` directory, the command is:

```
docker build -f Dockerfile-backend ..
```

i.e. it needs the root of the repo as context. When building from the root, it would be:

```
docker build -f docker/Dockerfile-backend .
```

Reply:
I would probably favor this:
```
docker build -f Dockerfile-backend ..
```

Reply:
Right. But that's for the CI, right? Should we merge this PR in the meantime, or should it come with the CI integration already?

Reply:
Ok, we can merge this for now.